### ISSUE TITLE
batchnorm_layer.c: Segmentation fault when training with CPU processing due to space not allocated

### ISSUE THREAD

**Issue by ptkahl** [AlexeyAB/darknet#7559]
File: src/batchnorm_layer.c
Issue: Variables assigned values without allocating space
Conditions: Occurs when training classifier if processing with CPU only ("GPU=0" in the Makefile) and "[batchnorm]" part of the neural network configuration

Details:
During classifier training with CPU only (i.e., no GPU), existing code attempts to assign values to the following float* variables (float arrays) during classifier training if "[batchnorm]" is part of the network configuration:
  layer.x
  layer.x_norm
  layer.mean_delta
  layer.variance_delta
An assignment attempt occurs implicitly through the use of copy_cpu() within a function defined in file batchnorm_layer.c, resulting in a runtime error ("segmentation fault").

Fix:
Dynamically allocate space for the variables in question when creating the batchnorm_layer in function make_batchnorm_layer() if training without GPU. For example, immediately after line 41 in file batchnorm_layer.c, add the following lines:

#ifndef GPU
    if (train) {
        layer.x = (float*)xcalloc(h * w * c * batch, sizeof(float));
        layer.x_norm = (float*)xcalloc(h * w * c * batch, sizeof(float));
        layer.mean_delta = (float*)xcalloc(c, sizeof(float));
        layer.variance_delta = (float*)xcalloc(c, sizeof(float));
    }
#endif // not GPU

**Comment by ptkahl** [comment]
Code update on 25 April 2021 appears to have fixed this issue in response to deoksangkim (#7619).
